# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/ArrayBufferObject.h
# Commit: 40c6d6eed7f8
# Full Hash: 40c6d6eed7f8ffd79078b778b6ceb8e78b28c2f0
# Author: Steve Fink <sfink@mozilla.com>
# Date: 2023-09-12 04:12:49
# Regressor Bug: 1690111
# File Overlap Count: 2
# Description:
#   Bug 1690111 - Implement JS::EnsureNonInlineArrayBufferOrView. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D176596
# ==============================================================================

diff -r 692f2a759573 -r 40c6d6eed7f8 js/src/vm/ArrayBufferObject.h
--- a/js/src/vm/ArrayBufferObject.h	Mon Sep 11 12:52:19 2023 +0000
+++ b/js/src/vm/ArrayBufferObject.h	Mon Sep 11 12:52:19 2023 +0000
@@ -416,6 +416,8 @@
     return true;
   }
 
+  static bool ensureNonInline(JSContext* cx, Handle<ArrayBufferObject*> buffer);
+
   // Detach this buffer from its original memory.  (This necessarily makes
   // views of this buffer unusable for modifying that original memory.)
   static void detach(JSContext* cx, Handle<ArrayBufferObject*> buffer);